splunk-otel-android/src/main/java/com/splunk/rum/NetworkMonitor.java

```diff
@@ -81,7 +81,7 @@ public void onAvailable(boolean deviceIsOnline, CurrentNetwork activeNetwork) {
                                 .startSpan();
                 // put these after span start to override what might be set in the
                 // RumAttributeAppender.
-                RumAttributeAppender.appendNetworkAttributes(available, activeNetwork);
+                available.setAllAttributes(activeNetwork.getNetworkAttributes());
```

Comment:
Are the `NetworkMonitor` unit tests lacking? A little surprised that this didn't impact a unit test.

Reply:
Yeah, we weren't setting the carrier there at all.